File: node/monitor.js
function setupMonitoring(client) {
    if (!process.env.PRINT_MONGODB_COMMANDS) return;
    client.on("commandStarted", e => console.log(`>> ${e.commandName} ${JSON.stringify(e.command)}`));
    client.on("commandSucceeded", e => console.log(`<< ${e.commandName} ${JSON.stringify(e.reply)}`));
}

module.exports = { setupMonitoring };

File: csharp/Monitor.cs
using System;
using MongoDB.Bson;
using MongoDB.Driver;
using MongoDB.Driver.Core.Events;

public static class Monitor {
    public static MongoClientSettings Configure(MongoClientSettings settings) {
        if (Environment.GetEnvironmentVariable("PRINT_MONGODB_COMMANDS") == null)
            return settings;
        settings.ClusterConfigurator = cb => {
            cb.Subscribe<CommandStartedEvent>(e =>
                Console.WriteLine($">> {e.CommandName} {e.Command.ToJson()}"));
            cb.Subscribe<CommandSucceededEvent>(e =>
                Console.WriteLine($"<< {e.CommandName} {e.Reply.ToJson()}"));
        };
        return settings;
    }
}
